src/main/java/com/crowdin/cli/commands/actions/DownloadReviewedSourcesAction.java

```diff
@@ -0,0 +1,256 @@
+import com.crowdin.cli.utils.PlaceholderUtil;
+import com.crowdin.cli.utils.Utils;
+import com.crowdin.cli.utils.console.ConsoleSpinner;
+import com.crowdin.client.sourcefiles.model.Branch;
+import com.crowdin.client.sourcefiles.model.BuildReviewedSourceFilesRequest;
+import com.crowdin.client.sourcefiles.model.FileInfo;
+import com.crowdin.client.sourcefiles.model.ReviewedStringsBuild;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.crowdin.cli.BaseCli.RESOURCE_BUNDLE;
+import static com.crowdin.cli.utils.console.ExecutionStatus.ERROR;
+import static com.crowdin.cli.utils.console.ExecutionStatus.WARNING;
+
+public class DownloadReviewedSourcesAction implements NewAction<PropertiesWithFiles, ProjectClient> {
+
+    private FilesInterface files;
+    private boolean noProgress;
+    private boolean plainView;
+    private String branchName;
+
+    private Outputter out;
+
+    public DownloadReviewedSourcesAction(
+            FilesInterface files, boolean noProgress, boolean plainView, String branchName
+    ) {
+        this.files = files;
+        this.noProgress = noProgress;
+        this.plainView = plainView;
+        this.branchName = branchName;
+    }
+
+
+    @Override
+    public void act(Outputter out, PropertiesWithFiles properties, ProjectClient client) {
+        this.out = out;
+        boolean isOrganization = PropertiesBeanUtils.isOrganization(properties.getBaseUrl());
+        if (!isOrganization) {
+            out.println(WARNING.withIcon(RESOURCE_BUNDLE.getString("error.only_enterprise")));
+            return;
+        }
+        if (!plainView && !properties.getPreserveHierarchy()) {
+            out.println(WARNING.withIcon(RESOURCE_BUNDLE.getString("message.download_sources.preserve_hierarchy_warning")));
+        }
+
+        CrowdinProjectFull project = ConsoleSpinner
+                .execute(out, "message.spinner.fetching_project_info", "error.collect_project_info",
+                        this.noProgress, this.plainView, client::downloadFullProject);
+
+        Long branchId = Optional.ofNullable(this.branchName)
+                .map(br -> project.findBranchByName(br)
+                        .orElseThrow(() -> new RuntimeException(RESOURCE_BUNDLE.getString("error.not_found_branch"))))
+                .map(Branch::getId)
+                .orElse(null);
+
+        BuildReviewedSourceFilesRequest request = new BuildReviewedSourceFilesRequest();
+        //branch id does not work properly
+//        request.setBranchId(branchId);
```

Comment:
What's wrong with the `branchId`?

Reply:
During my local testing I was not able to properly download build against specific `branchId`.  (download)[https://developer.crowdin.com/enterprise/api/v2/#operation/api.projects.strings.reviewed-builds.download.download] was returning 404

Reply:
So for now I am just not specifying it. The final zip archive will anyway contain files for different branches.